pkg/api/errors/errors.go

```diff
@@ -93,30 +93,32 @@ func FromObject(obj runtime.Object) error {
 }
 
 // NewNotFound returns a new error which indicates that the resource of the kind and the name was not found.
-func NewNotFound(kind, name string) error {
+func NewNotFound(qualifiedResource unversioned.GroupResource, name string) error {
```

Comment:
I'm fine with adding a resource field and returning it in that case.
Kind being empty is ok in error cases (practically, it already is).